include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -1486,10 +1486,11 @@ namespace internal
         }
     }
 
-    bool hp_functionality_enabled = false;
-    for (const auto &dh : dof_handler)
-      if (dh->get_fe_collection().size() > 1)
-        hp_functionality_enabled = true;
+    const bool hp_functionality_enabled =
+      std::any_of(dof_handlers.begin(), dof_handlers.end(), [](const auto &dh) {
+        return (dh->get_fe_collection().size() > 1);
```

Comment:
```suggestion
        return dh->has_hp_capabilities();
```
We store this information in a bool member. Since you changed these lines of code there, I would suggest to use the [`has_hp_capabilities`](https://www.dealii.org/developer/doxygen/deal.II/classDoFHandler.html#a9e3984e9ed172a9209a51c7e22e84768) here.

Reply:
Nice. So changed!